sympy/polys/matrices/domainmatrix.py

```diff
@@ -3256,6 +3256,52 @@ def lu(self):
         L, U, swaps = self.rep.lu()
         return self.from_rep(L), self.from_rep(U), swaps
 
+    def qr(self):
+        r"""
+        QR decomposition of the DomainMatrix.
+
+        Returns
+        =======
+
+        (Q, R)
+            Q is the orthogonal matrix and R is the upper triangular matrix
+            resulting from the QR decomposition of the DomainMatrix.
```

Comment:
There should be some explanation here that the Q matrix is not orthonormal. This is not what some would expect for a QR decomposition.